From 6c2af45ec8cff9b282d599dc098db0ca127bdc59 Mon Sep 17 00:00:00 2001
From: Renato Caldas <renato@calgera.com>
Date: Thu, 29 Jun 2023 13:59:11 +0100
Subject: [PATCH] libselinux: restore: drop the obsolete LSF transitional API.

The preferred way to enable LSF support on 32 bit systems is to define
_FILE_OFFSET_BITS=64 when building selinux.

Upstream-Status: Submitted [https://github.com/SELinuxProject/selinux/pull/401]

Signed-off-by: Renato Caldas <renato@calgera.com>
---
 src/selinux_restorecon.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/selinux_restorecon.c b/src/selinux_restorecon.c
index 38f10f1..5b3d035 100644
--- a/src/selinux_restorecon.c
+++ b/src/selinux_restorecon.c
@@ -436,7 +436,7 @@ static int filespec_add(ino_t ino, const char *con, const char *file,
 	file_spec_t *prevfl, *fl;
 	uint32_t h;
 	int ret;
-	struct stat64 sb;
+	struct stat sb;
 
 	__pthread_mutex_lock(&fl_mutex);
 
@@ -450,7 +450,7 @@ static int filespec_add(ino_t ino, const char *con, const char *file,
 	for (prevfl = &fl_head[h], fl = fl_head[h].next; fl;
 	     prevfl = fl, fl = fl->next) {
 		if (ino == fl->ino) {
-			ret = lstat64(fl->file, &sb);
+			ret = lstat(fl->file, &sb);
 			if (ret < 0 || sb.st_ino != ino) {
 				freecon(fl->con);
 				free(fl->file);
-- 
2.25.1

